### PR TITLE
fix(public-profile): harden training/cert projection follow-ups

### PR DESCRIPTION
**Pull request by fayazg**
## Summary

Non-blocking follow-ups from #1354 (public profile training/certification allow-list), hardening the S3 artifact projection:

- **Drift signal** — `projectTrainingActivities` now counts rows with an allow-listed Status but an unrecognized Type and logs a `logger.warning` (`dropped_for_unknown_type`), so an upstream myprofile vocabulary rename surfaces in Datadog instead of silently emptying the trainings rail. The allow-list is duplicated from myprofile with no shared contract, so this is our only drift signal.
- **Tighter epoch match** — the epoch-placeholder date guard now uses `startsWith('1970')` instead of `includes('1970')`, so a value that merely contains `1970` is no longer treated as the placeholder.
- **Documented certification divergences** — the certification filter's stricter-than-parity rules (drops an absent StartDate; `startsWith` epoch match) are now spelled out on the filter and the `PUBLIC_PROFILE_EPOCH_PLACEHOLDER` constant.
- **Consistent sort** — the certifications rail now sorts by Name to match the adjacent trainings rail.

Unit tests cover the drift signal (fires with the correct count / stays silent when drops are status-driven), the `startsWith` precision on both rails, and the certification Name sort.

Refs LFXV2-2933

### CODE REVIEW

File: apps/lfx-one/src/server/services/public-profile.service.ts [hunk 1]
```diff
@@ -109,29 +109,43 @@ function projectTechnicalContribution(value: unknown): PublicProfileTechnicalCon
...
+    // vocabulary likely drifted (allow-list duplicated here, no shared contract) — count it for Datadog.
+    if (statusAllowed && !typeAllowed) {
+      droppedForUnknownType++;
+    }
+    return statusAllowed && typeAllowed;
+  });
+  if (droppedForUnknownType > 0) {
+    logger.warning(req, 'project_public_profile', 'Dropped training rows with an unrecognized Type', {
+      dropped_for_unknown_type: droppedForUnknownType,
+    });
```
[the remark this answers is not in the document]
Comment (fayazg) [reply apps/lfx-one/src/server/services/public-profile.service.ts:147]: Good catch — confirmed and fixed in 29638060b. Verified against upstream (`linuxfoundation/myprofile` `profiles.plugin.js`): it defines only the type allow-list and never enumerates excluded types, so a real rename (e.g. `E-Learning` → `eLearning`) is still surfaced. The drift counter now increments only for a defined `Type` outside **both** the allow-list and a new `PUBLIC_PROFILE_TRAINING_TYPE_KNOWN_EXCLUDED` set (`Certification Exam`, `Subscription`, `Bundle`), so those normal Completed rows no longer emit the warning. Covered by a new known-excluded silence test.

File: apps/lfx-one/src/server/services/public-profile.service.ts [hunk 2]
```diff
@@ -109,29 +109,43 @@ function projectTechnicalContribution(value: unknown): PublicProfileTechnicalCon
...
+  let droppedForUnknownType = 0;
+  const kept = asRecordArray(value).filter((activity) => {
+    const status = pickString(activity['Status']);
+    const type = pickString(activity['Type']);
+    const statusAllowed = status !== undefined && PUBLIC_PROFILE_TRAINING_STATUS_ALLOWLIST.has(status);
+    const typeAllowed = type !== undefined && PUBLIC_PROFILE_TRAINING_TYPE_ALLOWLIST.has(type);
+    // Drift signal: an allow-listed Status with an unrecognized Type means the myprofile Type
+    // vocabulary likely drifted (allow-list duplicated here, no shared contract) — count it for Datadog.
+    if (statusAllowed && !typeAllowed) {
+      droppedForUnknownType++;
```
[the remark this answers is not in the document]
Comment (fayazg) [reply apps/lfx-one/src/server/services/public-profile.service.ts]: Fixed in 29638060b. Added `PUBLIC_PROFILE_TRAINING_TYPE_KNOWN_EXCLUDED` (`Certification Exam`, `Subscription`, `Bundle`) and now increment the drift counter only for a defined `Type` outside both the allow-list and that known-excluded set — so a Completed excluded row no longer trips the signal. Per your suggestion, the silence test now includes an allow-listed status with a known excluded type.

**Comment by fayazg** [comment]
## Review Feedback Addressed

Commit: 29638060b

### Changes Made
- **public-profile.constants.ts**: added `PUBLIC_PROFILE_TRAINING_TYPE_KNOWN_EXCLUDED` (`Certification Exam`, `Subscription`, `Bundle`) — the deliberately-dropped-but-normal training types (per cursor, copilot[bot])
- **public-profile.service.ts**: the drift counter now increments only for a defined `Type` outside **both** the allow-list and the known-excluded set, so normal Completed exam/subscription/bundle rows no longer emit `dropped_for_unknown_type`. Kept-row filtering is unchanged (per cursor, copilot[bot])
- **public-profile.service.spec.ts**: the drift test now includes a Completed `Bundle` row to prove it is *not* counted (count stays 2), plus a dedicated silence test for allow-listed status + known-excluded type

### Validation
Verified against upstream `linuxfoundation/myprofile` `profiles.plugin.js`, which defines only the type allow-list and never enumerates excluded types — so a genuine vocabulary rename (e.g. `E-Learning` → `eLearning`) is still surfaced by the drift signal.

### Threads Resolved
2 of 2 unresolved threads addressed in this iteration.

**Review by dealako — APPROVED** [review]
Hey @fayazg — reviewed this one with a three-way parallel pass (security/privacy, correctness/performance/tests, style/API/docs). Solid, well-scoped hardening PR: the drift-signal log, the `startsWith('1970')` precision fix, and the certification Name sort are all doing exactly what the description says, and the new unit tests cover the meaningful edge cases (drift count, silence on known-excluded/bad-status rows, startsWith precision on both rails, Name sort ordering).

**Revision history:** Cursor and Copilot both flagged the same real issue in the first round — the drift counter was going to fire on legitimate `Certification Exam`/`Subscription`/`Bundle` rows, drowning the real signal in noise. That's already resolved in `29638060` via `PUBLIC_PROFILE_TRAINING_TYPE_KNOWN_EXCLUDED`, verified against the upstream `myprofile` vocabulary, with a silence test added. Nothing new surfaced on top of that fix.

**Results:**
- 🔴 Blocking: 0
- 🟡 Minor: 0
- ⚪ Nit: 0
- ❔ Question: 0

No new security, correctness, performance, test-coverage, style, or documentation findings across the current diff.

✅ **Approved**
